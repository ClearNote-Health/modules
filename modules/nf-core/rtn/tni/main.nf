// TODO nf-core: A module file SHOULD only define input and output files as command-line parameters.
//               All other parameters MUST be provided using the "task.ext" directive, see here:
//               https://www.nextflow.io/docs/latest/process.html#ext
//               where "task.ext" is a string.
//               Any parameters that need to be evaluated in the context of a particular sample
//               e.g. single-end/paired-end data MUST also be defined and evaluated appropriately.
// TODO nf-core: Optional inputs are not currently supported by Nextflow. However, using an empty
//               list (`[]`) instead of a file can be used to work around this issue.

process RTN_TNI {
    debug true
    tag "{$expression_matrix.name}"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/bioconductor-rtn:2.26.0--r43hdfd78af_0':
        'biocontainers/bioconductor-rtn:2.26.0--r43hdfd78af_0' }"

    input:
    tuple val(meta), path(expression_matrix)

    output:
    path "tni.rds"                     , emit: tni
    path "tni_permutated.rds"          , emit: tni_perm
    path "tni_bootstrapped.rds"        , emit: tni_bootstrap
    path "tni_filtered.rds"            , emit: tni_filtered
    path "versions.yml"                , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''

    // TODO nf-core: It MUST be possible to pass additional parameters to the tool as a command-line string via the "task.ext.args" directive
    // TODO nf-core: If the tool supports multi-threading then you MUST provide the appropriate parameter
    //               using the Nextflow "task" variable e.g. "--threads $task.cpus"
    template 'rtn_tni.r'

    stub:
    def args = task.ext.args ?: ''

    """
    touch tni.rds
    touch tni_permutated.rds
    touch tni_bootstrapped.rds
    touch tni_filtered.rds

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bioconductor-rtn: \$(Rscript -e "suppressWarnings(library(RTN)); cat(as.character(packageVersion('RTN')))")
    END_VERSIONS
    """
}
