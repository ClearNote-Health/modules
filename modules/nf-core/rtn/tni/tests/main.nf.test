nextflow_process {

    name "Test Process RTN_TNI"
    script "../main.nf"
    process "RTN_TNI"

    tag "modules"
    tag "modules_nfcore"
    tag "rtn"
    tag "rtn/tni"

    test("musmusculus tni expression matrix") {
        when {
            process {
                """
                input[0] = [
                  [id:'test'],
                  file(params.modules_testdata_base_path + 'generic/tsv/expression.tsv', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
                // See https://nf-co.re/docs/contributing/tutorials/nf-test_assertions for more information and examples.
            )
        }

    }

    test("musmusculus tni expression matrix - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                  [id:'test'],
                  file(params.modules_testdata_base_path + 'generic/tsv/expression.tsv', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
            )
        }

    }

}
