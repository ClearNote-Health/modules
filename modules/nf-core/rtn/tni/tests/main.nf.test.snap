{
    "musmusculus tni expression matrix": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "tni.rds:md5,296d3df75ba6de468efb0cda91b19d96"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "tni_permutated.rds:md5,2b7a1256a1d322d581b6bd1552405a70"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        "tni_bootstrapped.rds:md5,6904abaccf03cafa2d2fefcbc816cc50"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test"
                        },
                        "tni_filtered.rds:md5,2f874bf4a325d463078bd5c8f43e17a7"
                    ]
                ],
                "4": [
                    "versions.yml:md5,47835259fd6d638971ddf96b6218ecea"
                ],
                "tni": [
                    [
                        {
                            "id": "test"
                        },
                        "tni.rds:md5,296d3df75ba6de468efb0cda91b19d96"
                    ]
                ],
                "tni_bootstrap": [
                    [
                        {
                            "id": "test"
                        },
                        "tni_bootstrapped.rds:md5,6904abaccf03cafa2d2fefcbc816cc50"
                    ]
                ],
                "tni_filtered": [
                    [
                        {
                            "id": "test"
                        },
                        "tni_filtered.rds:md5,2f874bf4a325d463078bd5c8f43e17a7"
                    ]
                ],
                "tni_perm": [
                    [
                        {
                            "id": "test"
                        },
                        "tni_permutated.rds:md5,2b7a1256a1d322d581b6bd1552405a70"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,47835259fd6d638971ddf96b6218ecea"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-05-22T12:13:02.01076"
    },
    "musmusculus tni expression matrix - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "tni.rds:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "tni_permutated.rds:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        "tni_bootstrapped.rds:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test"
                        },
                        "tni_filtered.rds:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "4": [
                    "versions.yml:md5,47835259fd6d638971ddf96b6218ecea"
                ],
                "tni": [
                    [
                        {
                            "id": "test"
                        },
                        "tni.rds:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "tni_bootstrap": [
                    [
                        {
                            "id": "test"
                        },
                        "tni_bootstrapped.rds:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "tni_filtered": [
                    [
                        {
                            "id": "test"
                        },
                        "tni_filtered.rds:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "tni_perm": [
                    [
                        {
                            "id": "test"
                        },
                        "tni_permutated.rds:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,47835259fd6d638971ddf96b6218ecea"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-05-22T12:15:01.324884"
    }
}